embedfs/embed_test.go

```diff
@@ -184,30 +178,23 @@ func TestReadDir(t *testing.T) {
 		wantErr bool
 	}{
 		{
-			name: "singleFile",
+			name: "testdata",
 			path: "testdata",
-			fs:   &singleFile,
-			want: []string{"empty.txt"},
+			fs:   testdata.GetTestData(),
+			want: []string{"empty.txt", "file1.txt", "file2.txt", "subdir"},
 		},
 		{
-			name:    "empty",
+			name:    "empty path",
 			path:    "",
-			fs:      &empty,
-			want:    []string{},
-			wantErr: true,
+			fs:      testdata.GetTestData(),
+			want:    []string{"testdata"},
+			wantErr: false,
 		},
 		{
-			name: "testdataDir w/ path",
-			path: "testdata",
-			fs:   &testdataDir,
-			want: []string{"empty.txt", "empty2.txt"},
-		},
-		{
-			name:    "testdataDir return no dir names",
-			path:    "",
-			fs:      &testdataDir,
-			want:    []string{},
-			wantErr: true,
+			name: "root path",
```

Comment:
This is a different sue case then the previous two, let's add them back up and combine with this new addition.